bin/run.py

```diff
@@ -41,34 +48,92 @@ def run(self, bar, *, interaction=None, submission_args=None):
+                max_duration = max(max_duration, result.duration)
+                if result.duration > self.problem.settings.timelimit:
+                    result.verdict = Verdict.TIME_LIMIT_EXCEEDED
+                    if tle_result is None:
+                        tle_result = result
+                        tle_result.pass_id = last_pass if self.problem.multipass else None
+                    else:
+                        tle_result.timeout_expired |= result.timeout_expired
+                elif result.status == ExecStatus.ERROR:
+                    result.verdict = Verdict.RUNTIME_ERROR
+                    if config.args.error:
+                        result.err = (
+                            'Exited with code ' + str(result.returncode) + ':\n' + result.err
+                        )
+                    else:
+                        result.err = 'Exited with code ' + str(result.returncode)
+
+                if interaction:
+                    data = self.in_path.read_text()
+                    if len(data) > 0 and data[-1] == '\n':
+                        data = data[:-1]
+                    data = data.replace('\n', '\n<')
+                    print('<', data, sep='', file=interaction)
+
+                    data = self.out_path.read_text()
+                    if len(data) > 0 and data[-1] == '\n':
+                        data = data[:-1]
+                    data = data.replace('\n', '\n>')
+                    print('>', data, sep='', file=interaction)
+
+                if result.verdict is None or self._continue_with_tle(
+                    result.verdict, result.timeout_expired
+                ):
+                    result = self._validate_output(bar)
+                    if result is None:
+                        bar.error(
+                            f'No output validators found for testcase {self.testcase.name}',
+                            resume=True,
+                        )
+                        result = ExecResult(
+                            None, ExecStatus.REJECTED, 0, False, None, None, Verdict.VALIDATOR_CRASH
+                        )
+                    elif result.status:
                         result.verdict = Verdict.ACCEPTED
+                        validate.sanity_check(self.out_path, bar, strict_whitespace=False)
                     elif result.status == ExecStatus.REJECTED:
                         result.verdict = Verdict.WRONG_ANSWER
+                        if nextpass and nextpass.is_file():
+                            bar.error(f'got WRONG_ANSWER but found nextpass.in', resume=True)
+                            result.verdict = Verdict.VALIDATOR_CRASH
                     else:
                         config.n_error += 1
                         result.verdict = Verdict.VALIDATOR_CRASH
 
+                    if result.verdict != Verdict.ACCEPTED:
+                        break
+                else:
```

Comment:
`if ...: ... else: break` => can you make it an early break and drop the `else:`?